device/devdoc/requirements/deviceclient_requirements.md

```diff
@@ -66,8 +66,17 @@ public sealed class DeviceClient
     public Task UpdateReportedPropertiesAsync(TwinCollection reportedProperties)
     public Task SetDesiredPropertyUpdateCallbackAsync(DesiredPropertyUpdateCallback callback, object userContext)
     
-    public async Task SetMethodHandlerAsync(string methodName, MethodCallback methodHandler, object userContext)
+    public async Task SendEventAsync(string outputName, Message message);
+    public async Task SendEventBatchAsync(string outputName, IEnumerable<Message> messages);
+    public async Task SetEventHandlerAsync(string inputName, MessageHandler messageHandler, object userContext);
+    public async Task SetEventDefaultHandlerAsync(MessageHandler messageHandler, object userContext);
+
+    internal async Task OnReceiveEventMessageCalled(EventMessageInternal eventMessageInternal);
 
+    public async Task<Twin> GetTwinAsync();
+    public Task<MethodResult> InvokeMethodAsync(string id, Method method);
```

Comment:
My bad, but I think we need to remove these InvokeMethod calls from here, for now at least.

Reply:
Easy. I'll do it...